packages/evm/contracts/BoostCore.sol

```diff
@@ -281,27 +256,80 @@ contract BoostCore is Ownable, ReentrancyGuard {
     {
         incentives = new AIncentive[](targets_.length);
         for (uint256 i = 0; i < targets_.length; i++) {
-            // Deploy the clone, but don't initialize until it we've preflighted
+            // Deploy the clone, but don't initialize until we've preflighted
             _checkTarget(type(AIncentive).interfaceId, targets_[i].instance);
 
             // Ensure the target is a base implementation (incentive clones are not reusable)
             if (!targets_[i].isBase) {
                 revert BoostError.InvalidInstance(type(AIncentive).interfaceId, targets_[i].instance);
             }
 
+            // Create the incentive instance
             incentives[i] = AIncentive(_makeTarget(type(AIncentive).interfaceId, targets_[i], false));
 
+            // Get the preflight data for the protocol fee and original disbursement
             bytes memory preflight = incentives[i].preflight(targets_[i].parameters);
             if (preflight.length != 0) {
-                // wake-disable-next-line reentrancy (false positive, entrypoint is nonReentrant)
+                // Protocol Fee disbursal
+                assert(budget_.disburse(_getFeeDisbursal(preflight)));
+                // Original disbursement call
                 assert(budget_.disburse(preflight));
             }
 
+            // Initialize the incentive instance after value has been trasnferred
             // wake-disable-next-line reentrancy (false positive, entrypoint is nonReentrant)
             incentives[i].initialize(targets_[i].parameters);
         }
     }
 
+    /// @notice Internal helper function to calculate the protocol fee and prepare the modified disbursal
+    /// @param preflight The encoded data for the original disbursement
+    /// @return The modified preflight data for the protocol fee disbursement
+    function _getFeeDisbursal(bytes memory preflight) internal view returns (bytes memory) {
+        // Decode the preflight data to extract the transfer details
+        ABudget.Transfer memory request = abi.decode(preflight, (ABudget.Transfer));
+
+        if (request.assetType == ABudget.AssetType.ERC20 || request.assetType == ABudget.AssetType.ETH) {
+            // Decode the fungible payload
+            ABudget.FungiblePayload memory payload = abi.decode(request.data, (ABudget.FungiblePayload));
+
+            // Calculate the protocol fee based on BIPS
+            uint256 feeAmount = (payload.amount * protocolFee) / FEE_DENOMINATOR;
+
+            // Create a new fungible payload for the protocol fee
+            ABudget.FungiblePayload memory feePayload = ABudget.FungiblePayload({amount: feeAmount});
+
+            // Modify the original request for the fee disbursal
+            request.data = abi.encode(feePayload);
+            request.target = address(this); // Set the target to BoostCore (this contract)
```

Comment:
don't we want this to go to the protocol fee receiver?

Reply:
Yeah definitely but it's going to be in another step.